File: src/main/java/com/dyrosoft/kvbparser/parser/StationLinesParserFuc.java
package com.dyrosoft.kvbparser.parser;

import com.dyrosoft.kvbparser.models.Line;
import com.google.common.base.Strings;
import com.google.common.collect.ImmutableList;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class StationLinesParserFuc extends AbstractRxHtmlParserFunc<ImmutableList<Line>> {

    @Override
    protected ImmutableList<Line> parse(final String html) {
        final ImmutableList.Builder<Line> builder = ImmutableList.builder();

        final Document document = Jsoup.parse(html);
        final Elements elements = document.getElementsByTag("a");
        for (final Element element : elements) {
            final String href = element.attr("href");
            if (Strings.isNullOrEmpty(href)) {
                continue;
            }

            final Pattern pattern = Pattern.compile("/german/hst/showline/(\\d+)/(\\d+)/");
            final Matcher matcher = pattern.matcher(href);
            if (!matcher.matches()) {
                continue;
            }

            final String temp = href.replace("/german/hst/showline/", "");
            final String id = temp.split("/")[1];

            builder.add(new Line(id));
        }

        return builder.build();
    }
}

File: src/main/java/com/dyrosoft/kvbparser/parser/StationLinesParserFuc.kt
package com.dyrosoft.kvbparser.parser

import com.dyrosoft.kvbparser.models.Line
import org.jsoup.Jsoup
import java.util.regex.Pattern

internal class StationLinesParserFuc : AbstractRxHtmlParserFunc<List<Line>>() {

    override fun parse(html: String): List<Line> {
        val list = mutableListOf<Line>()
        val document = Jsoup.parse(html)
        val elements = document.getElementsByTag("a")

        for (element in elements) {
            val href = element.attr("href")
            if (href.isNullOrEmpty()) {
                continue
            }

            val pattern = Pattern.compile("/german/hst/showline/(\\d+)/(\\d+)/")
            val matcher = pattern.matcher(href)
            if (!matcher.matches()) {
                continue
            }

            val temp = href.replace("/german/hst/showline/", "")
            val id = temp.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[1]

            list.add(Line(id))
        }

        return list.toList()
    }
}
